### ISSUE TITLE
Question rather than an issue - guild/server id

### ISSUE THREAD

**Issue by 0x4ndy** [serenity-rs/serenity#3072]
Hey, I've started playing with Serenity and I absolutely love it. Super easy to get started with a bot on discord (which is also something I'm new to). Can you tell me, how does the bot know which guild/server it is on? I don't see anything in the API, which makes me thing that the channel ID is the unique thing that identifies both the channel and the guild. Is that correct, or am I missing something from the API? 

Basically, what I'm trying to do is to see at any time, how many servers my bot is at. Thanks in advance!

**Comment by arqunis** [comment]
Hello. 

The primary way to find out how many guilds your bot has joined is to look into the cache, specifically the [`Cache::guilds()`](https://serenity-rs.github.io/serenity/current/serenity/cache/struct.Cache.html#method.guilds) function. You may access the cache through the [`Context::cache`](https://serenity-rs.github.io/serenity/current/serenity/client/struct.Context.html#structfield.cache) field.

Although, you are technically correct about a channel ID identifying both the channel, as well the guild it is from. Discord IDs are *globally* unique between entities of the same type (guilds, channels, roles, etc.), entities of different types (guilds != channels, channels != roles, guilds != roles, etc.), with very few exceptions where those IDs are shared (for instance, the @ everyone role has the same ID as the guild). You may retrieve data of a channel using [`ChannelId::to_channel`](https://serenity-rs.github.io/serenity/current/serenity/model/id/struct.ChannelId.html#method.to_channel), match it as a `GuildChannel`, and then get the guild ID it is from via the [`GuildChannel::guild_id`](https://serenity-rs.github.io/serenity/current/serenity/model/channel/struct.GuildChannel.html#structfield.guild_id) field.

**Comment by 0x4ndy** [comment]
Thanks @arqunis, that's very helpful. Is there also a way to capture the even of someone inviting my bot to their Discord guild?

**Comment by arqunis** [comment]
Watch for [`GUILD_CREATE` events](https://serenity-rs.github.io/serenity/current/serenity/client/trait.EventHandler.html#method.guild_create). If the `new` parameter is `true`, your bot was invited into a new guild.

**Comment by 0x4ndy** [comment]
Sweet, thanks!
